python/kserve/kserve/handlers/dataplane.py

```diff
@@ -0,0 +1,246 @@
+            https://kserve.github.io/website/0.9/modelserving/inference_api/#model-metadata
+
+        Raises: 'ModelNotFound' exception will be raised if the model with model_name is not found.
+
+        """
+
+        # Note: model versioning is not supported
+        model = self.get_model_from_registry(model_name)
+
+        if not isinstance(model, RayServeHandle):
+            input_types = model.get_input_types()
+            output_types = model.get_output_types()
+        else:
+            model_handle: RayServeHandle = model
+            input_types = await model_handle.get_input_types.remote()
+            output_types = await model_handle.get_output_types.remote()
+        return {
+            "name": model_name,
+            "platform": "",
+            "inputs": input_types,
+            "outputs": output_types
+        }
+
+    async def ready(self) -> bool:
+        """
+        Returns 'True' when all the loaded models are ready to receive requests.
+        Primarily meant to be used as kubernetes readiness check.
+
+        Returns:
+            True, when all registered models are ready
+            False, otherwise
+        """
+
+        models = self._model_registry.get_models().values()
+        is_ready = all([model.ready for model in models])
+        return is_ready
+
+    async def model_ready(self, model_name: str) -> bool:
+        """
+        Returns if the model specified by 'model_name' is ready
+
+        Args:
+            model_name (str): name of the model
+
+        Returns:
+            False, if the model is not ready or model is not found
+            True, otherwise
+        """
+        is_ready = self._model_registry.is_model_ready(model_name)
+        return is_ready
+
+    async def infer(
+            self,
+            model_name: str,
+            body: bytes,
+            headers: Dict[str, str] = None
+    ) -> Tuple[Union[str, bytes, Dict], Dict[str, str]]:
+        """
+        Performs inference on the specified model with the provided body and headers.
+        If the 'body' contains an encoded [CloudEvent](https://cloudevents.io/),
+        then it will be decoded and processed. The response body/headers will also be encoded as
+        CloudEvents.
+
+        Args:
+            model_name (str): Model name
+            body (bytes): The raw body of the received HTTP request.
+            headers (Dict[str,str]): The headers received for HTTP request
+
+        Returns:
+            A tuple with below elements:
+                - response: The inference result
+                - response_headers: Headers to construct the HTTP response
+        """
+
+        is_cloudevent = False
+        is_binary_cloudevent = False
+
+        if headers and has_binary_headers(headers):
+            is_cloudevent = True
+            is_binary_cloudevent = True
+            body = self.get_binary_cloudevent(body, headers)
+        else:
+            try:
+                body = json.loads(body)
```

Comment:
we might want to switch to use `rapidjson` for better serialization/deserialization performance especially for large inputs
https://pythonspeed.com/articles/faster-json-library/